### PR TITLE
EVG-19044: Add guidance to priority settings

### PR DESCRIPTION
**Pull request by minnakt**
EVG-19044

### Description
This PR surfaces a warning when users try to set their patch or task priority to any value greater than 50.

### Screenshots
Screenshot 2023-05-19 at 12 15 38 PM

### Testing
- Add tests in `SetPriority.test.tsx`

### CODE REVIEW

File: src/components/SetPriority/index.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            onChange={(e) => setPriority(parseInt(e.target.value, 10))}
+            onKeyPress={(e) => {
+              if (e.key === "Enter") {
+                onConfirm();
+                setOpen(false);
+              }
+            }}
+            size={16}
+            warning={showWarning}
+            type="number"
```
Comment (khelif96) [inline src/components/SetPriority/index.tsx:132]: I get these number increment buttons.
image

File: src/components/SetPriority/index.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+const { yellow } = palette;
+
+interface SetPriorityProps {
+  taskId?: string;
+  patchId?: string;
+  disabled?: boolean;
+  initialPriority?: number;
+}
+
+export const SetPriority: React.VFC<SetPriorityProps> = ({
```
Comment (khelif96) [inline src/components/SetPriority/index.tsx:28]: You can build some type safety into this with the following type

```ts
type SetPriorityProps = (
  | {
      patchId: string;
      taskId?: never;
    }
  | {
      taskId: string;
      patchId?: never;
    }
) & {
  initialPriority?: number;
  disabled?: boolean;
};
```

Comment (khelif96) [inline src/components/SetPriority/index.tsx:28]: nit. default export?

**Review by khelif96 — CHANGES_REQUESTED** [review]

File: src/components/SetPriority/index.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      </Popconfirm>
+    </>
+  );
+};
+
+const Container = styled.div`
+  width: 180px;
+`;
+
+const TextInput = styled(TextInputWithGlyph)<{ warning: boolean }>`
```
[the remark this answers is not in the document]
Comment (minnakt) [reply src/components/SetPriority/index.tsx]: changed the component!

**Review by khelif96 — APPROVED** [review]
Great Refactor!

**Pull request merged**